extensions/amp-render/amp-render.md

```diff
@@ -1,3 +1,183 @@
+2. **amp-state**: For initializing from `<amp-state>` data. See [Initialization from `<amp-state>`](#initialization-from-amp-state) for more details.
+3. **amp-script**: For using `<amp-script>` functions as the data source. See [Using `<amp-script>` as a data source](#using-amp-script-as-a-data-source) for more details.
+
+
+[tip type="important"]
+Your endpoint must implement the requirements specified in the [CORS Requests in AMP](https://www.ampproject.org/docs/fundamentals/amp-cors-requests) spec.
+[/tip]
+
+If fetching the data at the `src` URL fails, `<amp-render>` triggers a low-trust `fetch-error` event.
+
+The `src` attribute may be omitted if the `[src]` attribute exists. `[src]` supports URL and non-URL expression values.
+
+### `credentials`
+
+Defines a `credentials` option as specified by the [Fetch API](https://fetch.spec.whatwg.org/).
+
+-   Supported values: `omit`, `include`
+-   Default: `omit`
+
+To send credentials, pass the value of `include`. If this value is set, the response must follow the [AMP CORS security guidelines](https://amp.dev/documentation/guides-and-tutorials/learn/amp-caches-and-cors/amp-cors-requests/#cors-security-in-amp).
+
+Here's an example that specifies including credentials to display personalized content:
+
+```html
+<amp-render
+  credentials="include"
+  src="<%host%>/json/product.json?clientId=CLIENT_ID(myCookieId)"
+>
+  <template type="amp-mustache">
+    Your personal offer: ${{price}}
+  </template>
+</amp-render>
+```
+
+### `xssi-prefix`
+
+Causes `<amp-render>` to strip a prefix from the fetched JSON before parsing. This can be useful for APIs that include [security prefixes](http://patorjk.com/blog/2013/02/05/crafty-tricks-for-avoiding-xssi/) like `)]}` to help prevent cross site scripting attacks.
+
+For example, let's say we had an API that returned this response:
+
+```json
+)]}{ "items": ["value"] }
+```
+
+We could instruct `amp-render` to remove the security prefix like so:
+
+```html
+<amp-render xssi-prefix=")]}" src="https://example.com/data.json"></amp-render>
+```
+
+### `key`
+
+Defines the expression to locate the sub-object to be rendered within the response. For example, let's say we had an API that returned this response:
+
+```json
+{
+  "automobiles": {
+    "cars": {
+      "german": {
+        "make": "BMW",
+        "model": "M3"
+      },
+      "american": {
+        "make": "Tesla",
+        "model": "Model X"
+      }
+    }
+  }
+}
+```
+
+If we just want to display the German cars from the response, we can use the `key` attribute.
+
+```html
+<amp-render src="https://example.com/data.json" key="automobiles.cars.german">
+  <template type="amp-mustache">
+    {{make}} {{model}}
+  </template>
+</amp-render>
+```
+### `binding`
+
+For pages using `<amp-render>` that also use `amp-bind`, controls whether or not to block render on the evaluation of bindings (e.g. `[text]`) in rendered children.
```

Comment:
I'm not 100% sure what this means - do this attribute cause the rendering to wait until the binding evaluation is complete, or that an expression could block the rendering?

Reply:
Yes, depending on the value of this attribute, it can wait for binding evaluation and block rendering or not. This part is subject to change (we may not support binding="always" due to the performance impact).